src/pool.ts

```diff
@@ -689,6 +698,24 @@ class PoolImpl implements Pool {
     }
   }
 
+  async transaction<T>(
+    action: () => Promise<T>,
+    options?: TransactionOptions
+  ): Promise<T> {
+    let result: T;
+    const transaction = new Transaction(this, options);
+    await transaction.start();
+    try {
+      result = await action();
+      await transaction.commit();
+    } catch (err) {
+      await transaction.rollback();
+
+      throw err;
+    }
+    return result;
+  }
```

Comment:
Yeah, this needs to be fixed to explicitly acquire a connection from the pool, call `.transaction` on it, and then release the connection back to the pool.

Reply:
Will this be sufficient? I suspect not. If we want to support the same code api:

```
      await pool.transaction(async () => {
        await pool.execute(`
          INSERT ${typename} {
            name := 'Test Transaction'
          };
        `);

        // how can we be sure that the same connection is used in this second call to execute?
        await pool.execute(`
          INSERT ${typename} {
            name := 'Test Transaction'
          };
        `);
        await pool.execute("SELECT 1 / 0;");
      });
```

Wouldn't something like this be needed? (Looks a bit ugly and potentially confusing, though..)

```
      await pool.transaction(async (connectionInTransaction) => {
        await connectionInTransaction.execute(`
          INSERT ${typename} {
            name := 'Test Transaction'
          };
        `);

        await connectionInTransaction.execute(`
          INSERT ${typename} {
            name := 'Test Transaction'
          };
        `);
        await connectionInTransaction.execute("SELECT 1 / 0;");
      });
```

Reply:
This wouldn't work for sure:

```
      await pool.transaction(async () => {
        await pool.execute(`
```

so this is indeed the only way:

```
      await pool.transaction(async (con) => {
        await con.execute(`
```

We could, perhaps, prohibit using `pool` while the transaction closure runs... but I don't think we do that in our Python binding or in asyncpg and they seem to be OK.

@elprans what do you think?

Reply:
Actually, I don't see neither asyncpg nor edgedb-python implementing transaction helper for the Pool object. So we shouldn't do that for the JS binding. Just remove the `Pool.transaction` helper.

Reply:
Yes, `transaction` only makes sense on a specific connection.